buildSrc/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -61,7 +61,8 @@ public static SourceSetContainer getJavaSourceSets(Project project) {
     }
 
     public static <T> T maybeCreate(NamedDomainObjectContainer<T> collection, String name) {
-        return Optional.ofNullable(collection.findByName(name)).orElse(collection.create(name));
+        return collection.maybeCreate(name);
+        // return Optional.ofNullable(collection.findByName(name)).orElse(collection.create(name));
     }
 
     public static <T> T maybeCreate(NamedDomainObjectContainer<T> collection, String name, Action<T> action) {
```

Comment:
Can we have these delegate to `NamedDomainObjectContainer.maybeCreate()` as well?

Reply:
see above. all maybeCreate's have been removed